### PR TITLE
Small if err cleaning

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

File: api/client/build.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 			includes = append(includes, ".dockerignore", *dockerfileName)
 		}
 
-		if err = utils.ValidateContextDirectory(root, excludes); err != nil {
+		if err := utils.ValidateContextDirectory(root, excludes); err != nil {
```
Comment (LK4D4) [inline api/client/build.go]: Recently saw such check as `utils.ValidateContextDirectory(root, excludes) != nil` :) It is awful, but never thought that you can do this.

Comment (runcom) [inline api/client/build.go]: :open_mouth: cool, want me to change to that?

Comment (LK4D4) [inline api/client/build.go]: No :) I think it is bad for readability.

**Comment by LK4D4** [comment]
LGTM

File: integration-cli/docker_api_containers_test.go [hunk 2]
```diff
@@ -176,7 +176,7 @@ func (s *DockerSuite) TestContainerApiStartVolumesFrom(c *check.C) {
 		c.Fatal(out, err)
 	}
 
-	name := "testing"
+	name := "testing" + stringid.GenerateRandomID()
```
Comment (tiborvass) [inline integration-cli/docker_api_containers_test.go]: our convention usually is to use the test's name: `name := "testcontainerapistartvolumesfrom"` and btw I'd much rather have it named exactly like the test: TestContainerApiStartVolumesFrom, but that's not how we've done it so far.

**Comment by runcom** [comment]
updated @tiborvass

**Comment by tiborvass** [comment]
LGTM

**Comment by runcom** [comment]
rebasing

**Comment by tiborvass** [comment]
Needs another rebase, also we have to make sure that for each change, there is no `defer` that would have checked for `err`. That would terrible :( I'm rechecking now.

File: daemon/container.go [hunk 3]
```diff
@@ -149,8 +149,7 @@ func (container *Container) toDisk() error {
 		return err
 	}
 
-	err = ioutil.WriteFile(pth, data, 0666)
-	if err != nil {
+	if err := ioutil.WriteFile(pth, data, 0666); err != nil {
```
Comment (tiborvass) [inline daemon/container.go]: found another candidate for json.NewEncoder for another PR i guess.

**Comment by runcom** [comment]
@tiborvass I think I've addressed this but yes check again :)

**Comment by tiborvass** [comment]
Alright all `err` there are correctly fixed, there is no deferred err-check  issue :)

**Comment by tiborvass** [comment]
@runcom needs rebase sorry :(

**Comment by runcom** [comment]
@tiborvass rebased

**Comment by tiborvass** [comment]
re LGTM

**Pull request merged**